presto-native-execution/presto_cpp/main/types/PrestoToVeloxExpr.cpp

```diff
@@ -159,13 +159,46 @@ std::vector<TypedExprPtr> VeloxExprConverter::toVeloxExpr(
 }
 
 namespace {
+static const char* kVarchar = "varchar";
+
+/// Convert cast of varchar to substr if target type is varchar with max length.
+/// Throw an exception for cast of varchar to varchar with max length.
+std::optional<TypedExprPtr> convertCastToVarcharWithMaxLength(
+    const std::string& returnType,
+    const std::vector<TypedExprPtr>& args,
+    bool nullOnFailure) {
+  if (nullOnFailure) {
+    VELOX_NYI("TRY_CAST of varchar to {} is not supported.", returnType);
+  }
+
+  char* end;
+  const auto length =
```

Comment:
Please add a comment that you are trying to parse the length from a type string that looks like varchar[length] here.

Most of the code in this class uses Velox TypeParser https://github.com/facebookincubator/velox/blob/main/velox/type/parser/TypeParser.yy to parse returnType. Might be good to enhance it for varchar(n) instead of the custom code.

Reply:
Comment is added.

Reply:
This would require to have a VARCHAR(N) type available as it would need to be created as a result of the parsing.
I have this change in the prototype.